sdk/rust/oak_config_serializer/src/main.rs

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! An utility for serializing an Oak application configuration.
+//!
+//! Invoke with:
+//!
+//! ```shell
+//! cargo run --manifest-path=oak/server/rust/oak_config_serializer/Cargo.toml -- \
+//!     --input-file=<CONFIGURATION_FILE> \
+//!     --output-file=<OUTPUT_FILE>
+//! ```
+
+use anyhow::Context;
+use log::debug;
+use oak_abi::proto::oak::application::{
+    node_configuration::ConfigType, ApplicationConfiguration, NodeConfiguration,
+    WebAssemblyConfiguration,
+};
+use prost::Message;
+use std::{collections::HashMap, fs};
+use structopt::StructOpt;
+
+/// Command line options for the Oak Application Configuration Serializer.
+#[derive(StructOpt, Clone, Debug)]
+#[structopt(about = "Oak Application Configuration Serializer")]
+struct Opt {
+    #[structopt(long, help = "Input application configuration file in TOML format")]
+    input_file: String,
+    #[structopt(long, help = "Output file")]
+    output_file: String,
+}
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+struct Config {
+    name: String,
+    modules: HashMap<String, Module>,
+    #[serde(default)]
+    initial_node_configuration: InitialNodeConfig,
+}
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+enum Module {
+    #[serde(rename = "url")]
+    Url(String),
+    #[serde(rename = "path")]
+    Path(String),
+}
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+struct InitialNodeConfig {
+    wasm_module_name: String,
+    wasm_entrypoint_name: String,
+}
+
+impl Default for InitialNodeConfig {
+    fn default() -> Self {
+        Self {
+            wasm_module_name: "app".to_string(),
+            wasm_entrypoint_name: "oak_main".to_string(),
+        }
+    }
+}
```

Comment:
Does this mean that it can be omitted from the TOML files?

Reply:
Done